src/avt/Blueprint/avtConduitBlueprintDataAdaptor.C

```diff
@@ -1,85 +1,109 @@
+#include <conduit_blueprint.hpp>
+#include "avtConduitBlueprintLogging.h"
+#include "avtConduitBlueprintInfoWarningHandler.h"
 
 //-----------------------------------------------------------------------------
 // vtk includes
 //-----------------------------------------------------------------------------
-#include "vtkCharArray.h"
-#include "vtkUnsignedCharArray.h"
-#include "vtkShortArray.h"
-#include "vtkUnsignedShortArray.h"
-#include "vtkIntArray.h"
-#include "vtkUnsignedIntArray.h"
-#include "vtkLongArray.h"
-#include "vtkUnsignedLongArray.h"
-#include "vtkLongLongArray.h"
-#include "vtkUnsignedLongLongArray.h"
-#include "vtkFloatArray.h"
-#include "vtkDoubleArray.h"
-
-#include "vtkCellArray.h"
-#include "vtkCellType.h"
-#include "vtkCellData.h"
-#include "vtkIdTypeArray.h"
-#include "vtkPoints.h"
-#include "vtkPointData.h"
-#include "vtkRectilinearGrid.h"
-#include "vtkStructuredGrid.h"
-#include "vtkUnstructuredGrid.h"
-
-#include <vtkCell.h>
-#include <vtkLine.h>
-#include <vtkTriangle.h>
-#include <vtkHexahedron.h>
-#include <vtkQuad.h>
-#include <vtkTetra.h>
-#include <vtkPoints.h>
-
-//-----------------------------------------------------------------------------
-// visit includes
-//-----------------------------------------------------------------------------
-#include "InvalidVariableException.h"
-#include "Expression.h"
-#include "ExpressionList.h"
-#include "UnexpectedValueException.h"
-
-#include "avtMaterial.h"
-#include "avtMixedVariable.h"
-
-
-//-----------------------------------------------------------------------------
-// std lib includes
-//-----------------------------------------------------------------------------
-#include <string>
-
-//-----------------------------------------------------------------------------
-// conduit includes
-//-----------------------------------------------------------------------------
-#include "conduit.hpp"
-#include "conduit_blueprint.hpp"
-
-//-----------------------------------------------------------------------------
-// mfem includes
-//-----------------------------------------------------------------------------
-#include "mfem.hpp"
+#include <vtkCellData.h>
+#include <vtkDataArray.h>
+#include <vtkDoubleArray.h>
+#include <vtkUnstructuredGrid.h>
+#include <vtkCharArray.h>
+#include <vtkUnsignedCharArray.h>
+#include <vtkShortArray.h>
+#include <vtkUnsignedShortArray.h>
+#include <vtkIntArray.h>
+#include <vtkUnsignedIntArray.h>
+#include <vtkLongArray.h>
+#include <vtkUnsignedLongArray.h>
+#include <vtkFloatArray.h>
+#include <vtkCellArray.h>
+#include <vtkIdTypeArray.h>
+#include <vtkPointData.h>
+#include <vtkRectilinearGrid.h>
+#include <vtkStructuredGrid.h>
 
-
-//-----------------------------------------------------------------------------
-// bp visit plugin includes
-//-----------------------------------------------------------------------------
-#include "avtBlueprintLogging.h"
-
-using std::string;
 using namespace conduit;
 using namespace mfem;
```

Comment:
I see a potential problem: you are including `mfem.hpp` and `using namespace mfem`, yet this directory is included in the build regardless of whether or not FMEM_FOUND is true.

Reply:
The Blueprint plugin itself depends always on MFEM and Conduit.

The AVT Blueprint library should do this as well, but it doens't yet.